packages/edit-navigation/src/components/menu-locations-editor/menu-select-control.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { SelectControl } from '@wordpress/components';
+
+export default function MenuSelectControl( {
+	location,
+	availableMenuIds,
+	onSelectMenu,
+} ) {
+	const [ selectedMenuId, setSelectedMenuId ] = useState( undefined );
+
+	useEffect( () => {
+		setSelectedMenuId( location.menu );
+	}, [ location.menu ] );
+
+	return (
+		<SelectControl
+			options={ availableMenuIds }
+			value={ selectedMenuId }
+			onChange={ ( newMenuId ) => {
+				onSelectMenu( selectedMenuId, {
+					newLocation: location.name,
+					newMenuId: parseInt( newMenuId ),
```

Comment:
Is `parseInt` necessary? Similar to my earlier comment, it's better to ensure that the data structure always contains values with the correct type rather than having to coerce values at the time of consumption.

Reply:
Yeah, not needed apparently. But I can swear at some point I got into trouble with the API because of it. Can't repro now though.